csi/controller.go

```diff
@@ -260,6 +272,37 @@ func osdVolumeContext(v *api.Volume) map[string]string {
 	}
 }
 
+func getPVCMetadata(params map[string]string) (map[string]string, error) {
+	metadata := make(map[string]string)
+
+	metadata[intreePvcNameKey] = params[osdPvcNameKey]
+	metadata[intreePvcNamespaceKey] = params[osdPvcNamespaceKey]
+	decodedLabels := make(map[string]string)
+	decodedAnnotations := make(map[string]string)
+	if len(params[osdPvcAnnotationsKey]) > 0 || len(params[osdPvcLabelsKey]) > 0 {
```

Comment:
Why have an OR in the if. If we are going to use an `if` then it should be for each one (a function maybe?)

Reply:
Broke this down into a function.